src/cargo/ops/cargo_compile.rs

```diff
@@ -116,12 +116,19 @@ pub enum CompileFilter<'a> {
 
 pub fn compile<'a>(ws: &Workspace<'a>, options: &CompileOptions<'a>)
                    -> CargoResult<ops::Compilation<'a>> {
+    compile_with_exec(ws, options, &mut DefaultExecutor)
+}
+
+pub fn compile_with_exec<'a, E: Executor>(ws: &Workspace<'a>,
+                                          options: &CompileOptions<'a>, 
+                                          exec: &mut E)
```

Comment:
Hm so `Executor: Clone` could throw a wrench into this, but perhaps that could be jerry-rigged with something like:

```rust
fn clone(&self) -> Box<Exectuor>;
```

(or something else manual like that)